clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -613,15 +613,46 @@ public void testOffsetFetchRequestErroredRequests(final Errors error, final bool
         }
     }
 
+    @ParameterizedTest
+    @MethodSource("offsetFetchExceptionSupplier")
+    public void testOffsetFetchRequestTimeoutRequests(final Errors error,
+                                                      final boolean isRetriable) {
+        CommitRequestManager commitRequestManager = create(true, 100);
+        when(coordinatorRequestManager.coordinator()).thenReturn(Optional.of(mockedNode));
+
+        Set<TopicPartition> partitions = new HashSet<>();
+        partitions.add(new TopicPartition("t1", 0));
+        List<CompletableFuture<Map<TopicPartition, OffsetAndMetadata>>> futures = sendAndVerifyDuplicatedOffsetFetchRequests(
+                commitRequestManager,
+                partitions,
+                1,
+                error);
+
+        if (isRetriable) {
```

Comment:
Do you also plan to use here `error.exception() instanceof RetriableException`?

Reply:
this one should be like this I would say, because this is the fetch (where we so have errors that from the exception definition are retriables, but we don't consider them retriable)

Reply:
If the exceptions are just a few, wouldn't it be better for readability, to call them out explicitly here in the test?

Reply:
yes, that makes sense to me. In the end it's a good point that we only have 2 exceptions for retriables that are not considered retriable on the fetch path: unknown_topic and request timeout, so we could simply maybe have a helper isRetriableOnFetch that would return true if error.exception() instanceof RetriableException && !unknown_topic && !request_timeout, something like it. Up to you @kirktrue, but makes sense to me to get rid of the arg to maintain and be explicit as Bruno suggested.

Reply:
This is what I have that is needed to work:

```java
private boolean isRetriableOnOffsetFetch(Errors error) {
    return error == Errors.NOT_COORDINATOR || error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.COORDINATOR_NOT_AVAILABLE;
}
```

Does that seem correct, @lianetm?

Reply:
After removing the boolean flags from `offsetFetchExceptionSupplier()`, it became identical to `offsetCommitExceptionSupplier()`. So I refactored things a bit further and renamed `offsetCommitExceptionSupplier()` as `offsetExceptionSupplier()` for use by the relevant parameterized tests.

Reply:
those are right, but you're missing `UnstableOffsetCommitException`, that is retriable and thrown on fetch [here](https://github.com/apache/kafka/blob/896af1b2f2f2a7d579e0aef074bcb2004c0246f2/clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java#L1073).

Reply:
If I add that to `offsetExceptionSupplier()` and `isRetriableOnOffsetFetch()`, the relevant tests fail 🤔

That error is being tested in `testOffsetFetchRequestPartitionDataError()` via its supplier `partitionDataErrorSupplier()`.